### DISCUSSION TITLE
lr scheduler can't monitor val metric

### DISCUSSION THREAD

**Discussion by morestart** [Lightning-AI/pytorch-lightning#10599] [General]
this is my code:
```Python

class DetModel(pl.LightningModule):

    def forward(self, x):
        features = self.encoder(x)
        features = self.neck(features)
        features = self.head(features)

        return features

    def training_step(self, batch, batch_idx):
        data = batch
        output = self.forward(data['img'])
        loss_dict = self.loss_func(output, batch)

        self.log(name=self.train_loss_name, value=loss_dict['loss'])
        self.log(name='shrink_maps', value=loss_dict['loss_shrink_maps'])
        self.log(name='threshold_maps', value=loss_dict['loss_threshold_maps'])
        self.log(name='binary_maps', value=loss_dict['loss_binary_maps'])

        return loss_dict['loss']

    def validation_step(self, batch, batch_idx):
        data = batch

        output = self.forward(data['img'])
        boxes, scores = self.postprocess(output.cpu().numpy(), batch['shape'])
        raw_metric = self.metric(batch, (boxes, scores))

        return raw_metric

    def validation_epoch_end(self, outputs):
        metric = self.metric.gather_measure(outputs)
        self.log('recall', value=metric['recall'].avg)
        self.log('precision', value=metric['precision'].avg)
        self.log('hmean', value=metric['fmeasure'].avg)
        return {'hmean': metric['fmeasure'].avg}

    def configure_optimizers(self):
        optimizer = get_optimizer(self.parameters(), self.optimizer_name, self.lr)
        lr_scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(optimizer, mode='max')
        # return {'optimizer': optimizer, 'lr_scheduler': lr_scheduler, "monitor": 'hmean'}
        return (
            [
                optimizer
            ],
            [
                {
                    'scheduler': lr_scheduler,
                    'interval': 'step',
                    'frequency': 1,
                    'reduce_on_plateau': False,
                    'monitor': 'heman',
                }
            ]
        )

```

when i run this code, i got:
```
pytorch_lightning.utilities.exceptions.MisconfigurationException: ReduceLROnPlateau conditioned on metric heman which is not available. Available metrics are: ['train_loss', 'shrink_maps', 'threshold_maps', 'binary_maps']. Condition can be set using `monitor` key in lr scheduler dict

```

**Comment by tchaton** [comment]
Dear @morestart.

Mind trying this ?

```py

    def configure_optimizers(self):
        optimizer = get_optimizer(self.parameters(), self.optimizer_name, self.lr)
        lr_scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(optimizer, mode='max')
        # return {'optimizer': optimizer, 'lr_scheduler': lr_scheduler, "monitor": 'hmean'}
        return (
            [
                optimizer
            ],
            [
                {
                    'scheduler': lr_scheduler,
                    'interval': 'step',
                    'monitor': 'heman',
                }
            ]
        )
```

**Reply by morestart** [reply]
not work, same error
```
pytorch_lightning.utilities.exceptions.MisconfigurationException: ReduceLROnPlateau conditioned on metric heman which is not available.
 Available metrics are: ['train_loss', 'shrink_maps', 'threshold_maps', 'binary_maps'].
 Condition can be set using `monitor` key in lr scheduler dict
```

**Reply by akihironitta** [reply]
@morestart I believe fixing the typo should resolve your issue:

```diff
-                    'monitor': 'heman',
+                    'monitor': 'hmean',
```

**Reply by morestart** [reply]
> @morestart I believe fixing the typo should resolve your issue:
> 
> ```diff
> -                    'monitor': 'heman',
> +                    'monitor': 'hmean',
> ```

emmm,It was a stupid mistake!
thanks for your help~
